aidb/engine/limit_engine.py

```diff
@@ -42,11 +43,21 @@ async def _execute_limit_query(self, query: Query):
 
     # TODO: rewrite query, use full scan to execute query
     bound_service_list = query.inference_engines_required_for_query
-    for index, _ in sorted_list:
+
+    limit_engine_batch_size = LIMIT_ENGINE_BATCH_SIZE_DEFAULT
```

Comment:
if this is never overriden just use the default

Reply:
Hmm, I think it would be more effective to avoid using a default value and instead utilize the smallest preferred_batch_size across all bound_services.